src/components/nutrition/food/custom-food.tsx

```diff
@@ -139,9 +139,9 @@ export function CustomFoodFields() {
           loading={!allUnits}
           autocompleteProps={{
             getOptionKey: (option) => option.id,
-            isOptionEqualToValue: (option, value) => option.id === value.id,
+            isOptionEqualToValue: (option, value) => value && option.id === value.id,
           }}
```

Comment:
This seems to work a bit better for keeping the form fields on one line

```suggestion
            isOptionEqualToValue: (option, value) => value && option.id === value.id,
            sx: { flex: 1 },
          }}
```